### PR TITLE
refs #302 use oAuth config from vuex

### PR DESCRIPTION
**Pull request by tempelgogo**
* adjusted sample config, containing the oAuth api url & token
* inject config into vuex
* use oAuth config from vuex
* mutation names adopted

### CODE REVIEW

File: src/store/auth.js [hunk 1]
```diff
@@ -1,59 +1,67 @@
...
+  },
+  getters: {
+    isAuthenticated: state => {
+      return state.isAuthenticated
+    }
   },
-
   mutations: {
-    isAuthenticated (state, payload) {
+    IS_AUTHENTICATED(state, payload) {
```
Comment (DeepDiver1975) [inline src/store/auth.js]: Is this some vuex naming convention? Uppercase functions look :ghost:

File: src/services/auth.js [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
-      clientId: 'Cxfj9F9ZZWQbQZps1E1M0BszMz6OOFq3lxjSuc8Uh4HLEYb9KIfyRMmgY5ibXXrU',
-      url: 'http://deepdiver:8080/index.php/apps/oauth2/api/v1/token',
-      authorizationEndpoint: 'http://deepdiver:8080/index.php/apps/oauth2/authorize',
-      responseType: 'token',
-      responseParams: {
-        code: 'code',
-        clientId: 'client_id',
-        redirectUri: 'redirect_uri'
+export function initVueAuthenticate (config) {
+  if (config) {
```
Comment (DeepDiver1975) [inline src/services/auth.js]: in case config is missing we need to render some error page or something ....

File: src/services/auth.js [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
+    return new VueAuthenticate(Axios, {
+      providers: {
+        oauth2: {
+          name: 'ownCloud',
+          clientId: config.clientId || '',
```
Comment (DeepDiver1975) [inline src/services/auth.js]: I think there are no reasonable default - if not set -> :boom:

File: src/store/auth.js [hunk 4]
```diff
@@ -1,59 +1,67 @@
...
           displayname: res['display-name'],
           email: Object.keys(res.email).length === 0 ? '' : res.email
         })
       })
     },
 
-    authenticate (context, payload) {
+    authenticate(context, payload) {
+      // TODO shouldn't we use vue-authenticate global & inject it into vue ?
+      vueAuthInstance = initVueAuthenticate(context.rootState.config.auth)
```
Comment (DeepDiver1975) [inline src/store/auth.js]: context.rootState.config.auth is undefined ....

screenshot from 2018-11-05 14-43-07

File: src/default.js [hunk 5]
```diff
@@ -73,6 +73,9 @@ Vue.component('drop', Drop);
                 redirect : to => arguments[0].navItems[0].route
             }];
 
+            // inject custom config into vuex base state
+            store.state.config.auth = config.data.auth
```
Comment (DeepDiver1975) [inline src/default.js]: ah - config.data is a string - this is not yet parsed json .... because of that auth does not work ...

Comment (DeepDiver1975) [reply src/default.js]: local issue with my config.json -> ignire

File: src/store/auth.js (hunk 1 above)
Comment (tempelgogo) [reply src/store/auth.js]: internal usage:
https://github.com/serv3r/phoenix/blob/master/apps/files/src/store.js
external recommendation:
https://vuejs.org/v2/style-guide/#Non-flux-state-management-use-with-caution

File: src/services/auth.js (hunk 2 above)
Comment (tempelgogo) [reply src/services/auth.js]: Is there already a route or component for the error page?

File: src/services/auth.js (hunk 3 above)
Comment (tempelgogo) [reply src/services/auth.js]: https://github.com/owncloud/phoenix/pull/303#discussion_r230759679

File: src/services/auth.js (hunk 2 above)
Comment (DeepDiver1975) [reply src/services/auth.js]: we don't have something like this - I'll open a issue for this ... THX

**Pull request merged**